src/controllers/midi/midimessage.h

```diff
@@ -223,10 +221,13 @@ struct MidiInputMapping {
                         options == other.options &&
                         std::get<ConfigKey>(control) == std::get<ConfigKey>(other.control) &&
                         description == other.description;
-            } else if constexpr (std::is_same_v<T, QJSValue>) {
+            } else if constexpr (std::is_same_v<T, std::shared_ptr<QJSValue>>) {
+                const std::shared_ptr<QJSValue> otherControl =
+                        std::get<std::shared_ptr<QJSValue>>(other.control);
+                const std::shared_ptr<QJSValue> thisControl =
+                        std::get<std::shared_ptr<QJSValue>>(control);
```

Comment:
if you make these a reference, we don't encounter the reference-counting overhead. In this case its possible because we don't take ownership and the references don't outlive the shared_ptr they're obtained from.

Reply:
You mean like this :

```suggestion
                const std::shared_ptr<QJSValue>& otherControl =
                        std::get<std::shared_ptr<QJSValue>>(other.control);
                const std::shared_ptr<QJSValue>& thisControl =
                        std::get<std::shared_ptr<QJSValue>>(control);
```

Reply:
Or more like this?

```suggestion
                const auto& otherControl = *std::get<std::shared_ptr<QJSValue>>(other.control);
                const auto& thisControl = *std::get<std::shared_ptr<QJSValue>>(control);
```

Reply:
The second one compiles and run correctly. I prefer it tbh. So I added this code to my controller:

```js
for(let i = 0; i < 1000; i++){
    midi.makeInputHandler(0x90, 0x43, () => {
        console.error("Without variable");
    });
    gc();
    this.inputHandlers.push(
        midi.makeInputHandler(0x90, 0x43, () => {
            console.error("With variable");
        })
    );
}
```

None of the handlers were reclaimed. It's been running for an hour now. The handler still correctly react to the buttons press.

Reply:
> The second one compiles and run correctly. I prefer it tbh

But it's incorrect... at least it doesn't avoid the unnecessary reference-count manipulation... I also find it a little weird because its quite unidiomatic. Out of curiosity, why do you prefer it?

Reply:
> None of the handlers were reclaimed.

So the one without variable didn't either? I guess that's possible, it's up to the gc to decide when to reclaim... Still, it would probably crash once it does reclaim any of these handlers (without the shared_ptr we added).

Reply:
> So the one without variable didn't either?

Nope. Everything worked perfectly until the end of the mix.